packages/@react-aria/table/src/useTableColumnHeader.ts

```diff
@@ -80,6 +80,19 @@ export function useTableColumnHeader<T>(props: AriaTableColumnHeaderProps<T>, st
 
   let descriptionProps = useDescription(sortDescription);
 
+  let shouldDisableFocus = (state.collection.size === 0 &&
+    state.collection.body.props.loadingState !== 'loading' &&
+    state.collection.body.props.loadingState !== 'loadingMore');
+  // delay disabling focus so we get a blur event on the cell to clean up the focus ring
+  let [disableFocus, setDisableFocus]  = useState(shouldDisableFocus);
+
+  useLayoutEffect(() => {
+    if (shouldDisableFocus && !disableFocus && ref.current === document.activeElement) {
+      ref.current?.blur();
```

Comment:
Ideally we've move focus somewhere meaningful, but I'm not sure where that is yet